src/Kafka/ConsumerGroups/components/ConsumerGroupDetail.tsx

```diff
@@ -0,0 +1,198 @@
+              <ConsumerGroupPopover
+                title={t("consumerGroup.partitions_with_lag_name")}
+                description={t("consumerGroup.partitions_with_lag_description")}
+              />
+            </Text>
+            <div>
+              <Text component={TextVariants.h2}>
+                {consumerGroupDetail &&
+                  consumerGroupDetail.consumers.reduce(function (prev, cur) {
+                    return prev + (cur.lag > 0 ? 1 : 0);
+                  }, 0)}
+              </Text>
+            </div>
+          </FlexItem>
+          <FlexItem>
+            <Text component={TextVariants.h4}>{t("consumerGroup.state")}</Text>
+            <div>
+              <Text component={TextVariants.h2}>
+                {displayConsumerGroupState[consumerGroupDetail.state]}
+              </Text>
+            </div>
+          </FlexItem>
+        </Flex>
+      </TextContent>
+      <TableComposable
+        aria-label={t("consumerGroup.consumer_group_info_table_aria")}
+        variant={TableVariant.compact}
+      >
+        <Thead noWrap>
+          <Tr>
+            {consumerGroupByTopic
+              ? [
+                  <Th key={columnNames.partition} width={20}>
+                    {columnNames.partition}
+                  </Th>,
+                  <Th key={columnNames.consumer_id} width={20}>
+                    {columnNames.consumer_id}
+                  </Th>,
+                  <Th key={columnNames.current_offset} width={20}>
+                    {columnNames.current_offset}
+                  </Th>,
+                  <Th key={columnNames.log_end_offset} width={20}>
+                    {columnNames.log_end_offset}
+                  </Th>,
+                  <Th key={columnNames.offset_lag} width={20}>
+                    {columnNames.offset_lag}
+                  </Th>,
+                ]
+              : [
+                  <Th key={columnNames.topic} width={20}>
+                    {columnNames.topic}
+                  </Th>,
+                  <Th key={columnNames.partition} width={20}>
+                    {columnNames.partition}
+                  </Th>,
+                  <Th key={columnNames.consumer_id} width={20}>
+                    {columnNames.consumer_id}
+                  </Th>,
+                  <Th key={columnNames.current_offset} width={20}>
+                    {columnNames.current_offset}
+                  </Th>,
+                  <Th key={columnNames.log_end_offset} width={20}>
+                    {columnNames.log_end_offset}
+                  </Th>,
+                  <Th key={columnNames.offset_lag} width={20}>
+                    {columnNames.offset_lag}
+                  </Th>,
+                ]}
+          </Tr>
+        </Thead>
+        <Tbody>
+          {(consumerGroupDetail &&
+            consumerGroupDetail.consumers.map((consumergroup) => {
+              return (
+                <Tr key={consumergroup.groupId}>
+                  {consumerGroupByTopic
```

Comment:
There are 3 of these ifs in the code. It makes the code difficult to follow. 

I'd create two different components and have the parent component use one or another depending on the need.